components/core/DataViewV0.js

```diff
@@ -0,0 +1,824 @@
+    color: ${Constants.system.brand};
+  }
+`;
+
+const STYLES_ICON_BOX_BACKGROUND = css`
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+  height: 25px;
+  width: 25px;
+  cursor: pointer;
+  background-color: rgba(255, 255, 255, 0.75);
+  border-radius: 3px;
+  position: absolute;
+  bottom: 8px;
+  right: 8px;
+`;
+
+const STYLES_ACTION_BAR = css`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  box-shadow: 0 0 0 1px ${Constants.system.lightBorder} inset,
+    0 0 4px 2px ${Constants.system.shadow};
+  border-radius: 4px;
+  padding: 12px 32px;
+  box-sizing: border-box;
+  background-color: ${Constants.system.foreground};
+  position: fixed;
+  bottom: 12px;
+  width: calc(100vw - ${Constants.sizes.sidebar}px + 32px);
+  max-width: ${Constants.sizes.desktop}px;
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    width: calc(100vw - 48px);
+  }
+`;
+
+const STYLES_RIGHT = css`
+  flex-shrink: 0;
+  display: flex;
+  align-items: center;
+`;
+
+const STYLES_LEFT = css`
+  width: 100%;
+  min-width: 10%;
+  display: flex;
+  align-items: center;
+`;
+
+const STYLES_FILES_SELECTED = css`
+  font-family: ${Constants.font.semiBold};
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    display: none;
+  }
+`;
+
+const STYLES_COPY_INPUT = css`
+  pointer-events: none;
+  position: absolute;
+  opacity: 0;
+`;
+
+const STYLES_IMAGE_GRID = css`
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(214px, 1fr));
+  margin: 0 -27px;
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    display: grid;
+    grid-template-columns: 1fr 1fr;
+    margin: 0px -12px;
+  }
+`;
+
+const STYLES_IMAGE_BOX = css`
+  width: 160px;
+  height: 160px;
+  margin: 27px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  box-shadow: 0px 0px 0px 1px ${Constants.system.lightBorder} inset,
+    0 0 40px 0 ${Constants.system.shadow};
+  cursor: pointer;
+  position: relative;
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    width: 144px;
+    height: 144px;
+    margin: 12px auto;
+  }
+`;
+
+const STYLES_MOBILE_HIDDEN = css`
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    display: none;
+  }
+`;
+
+const delay = (ms) => new Promise((resolve) => window.setTimeout(resolve, ms));
+
+let mounted = false;
+
+export default class DataView extends React.Component {
+  _mounted = false;
+
+  state = {
+    menu: null,
+    loading: {},
+    checked: {},
+    view: "grid",
+    viewLimit: 5,
+    scrollDebounce: false,
+  };
+
+  async componentDidMount() {
+    if (!mounted) {
+      mounted = true;
+      window.addEventListener("remote-data-deletion", this._handleDataDeletion);
+      window.addEventListener("remote-slate-object-remove", this._handleRemoteSlateObjectRemove);
+      window.addEventListener("remote-slate-object-add", this._handleRemoteSlateObjectAdd);
+    }
+
+    window.addEventListener("scroll", this._handleScroll);
```

Comment:
Move this inside the `if(!mounted)` above (the purpose of that is so that it isn't called twice when there is a double mount) and change `this._handleScroll` to `this._handleCheckScroll` (otherwise it isn't calling the debounce at all)